test/parallel-benchmark/mzcompose.py

```diff
@@ -0,0 +1,1190 @@
+                    + [
+                        ClosedLoop(
+                            action=StandaloneQuery(
+                                "SELECT * FROM mv_sum",
+                                conn_infos["materialized"],
+                                strict_serializable=True,
+                            ),
+                        )
+                        for i in range(10)
+                    ],
+                ),
+            ],
+            guarantees={
+                "SELECT * FROM mv_sum (standalone)": {"qps": 15, "max": 4000},
+            },
+        )
+
+
+class PgReadReplicaRTR(Scenario):
+    def __init__(self, c: Composition, conn_infos: dict[str, PgConnInfo]):
+        self.init(
+            [
+                TdPhase(
+                    """
+                    > DROP SECRET IF EXISTS pgpass CASCADE
+                    > CREATE SECRET pgpass AS 'postgres'
+                    > CREATE CONNECTION pg TO POSTGRES (
+                        HOST postgres,
+                        DATABASE postgres,
+                        USER postgres,
+                        PASSWORD SECRET pgpass
+                      )
+
+                    $ postgres-execute connection=postgres://postgres:postgres@postgres
+                    ALTER USER postgres WITH replication;
+                    CREATE TABLE t2 (f1 INTEGER);
+                    ALTER TABLE t2 REPLICA IDENTITY FULL;
+                    CREATE PUBLICATION mz_source3 FOR ALL TABLES;
+
+                    > CREATE SOURCE mz_source3
+                      FROM POSTGRES CONNECTION pg (PUBLICATION 'mz_source3')
+                      FOR ALL TABLES;
+
+                    > CREATE MATERIALIZED VIEW mv_sum AS
+                      SELECT COUNT(*) FROM t2;
+
+                    > CREATE DEFAULT INDEX ON mv_sum;
+                    """
+                ),
+                LoadPhase(
+                    duration=360,
+                    actions=[
+                        OpenLoop(
+                            action=StandaloneQuery(
+                                "INSERT INTO t2 VALUES (1)",
+                                conn_infos["postgres"],
+                            ),
+                            dist=Periodic(per_second=100),
+                        ),
+                        OpenLoop(
+                            action=StandaloneQuery(
+                                "SET REAL_TIME_RECENCY TO TRUE; SELECT * FROM mv_sum",
+                                conn_infos["materialized"],
+                                strict_serializable=False,
+                            ),
+                            dist=Periodic(per_second=125),
+                        ),
+                    ],
+                ),
+            ],
+            guarantees={
+                # TODO(def-): Lower max when RTR becomes more performant
+                "SET REAL_TIME_RECENCY TO TRUE; SELECT * FROM mv_sum (standalone)": {
+                    "qps": 50,
```

Comment:
qps is queries per second, right? Why is the value higher compared to `PgReadReplica`?
Resp. why "Lower max when RTR becomes more performant"? Am I misunderstanding that a higher value implies higher performance?

Reply:
They run different things. The PgReadReplica scenario is closed loop with 10 threads in parallel, while the PgReadReplicaRTR is open loop with 125 queries per second.